spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseExpireSnapshotsSparkAction.java

```diff
@@ -170,13 +177,15 @@ public BaseExpireSnapshotsSparkAction deleteWith(Consumer<String> newDeleteFunc)
         expireSnapshots = expireSnapshots.retainLast(retainLastValue);
       }
 
+      List<Snapshot> expired = expireSnapshots.apply();
```

Comment:
There is no guarantee that `apply` followed by `commit` will produce the same result, so this is unsafe. It may work most of the time, but there will probably be leaks when `expired` here doesn't contain a snapshot that was actually removed. That's why we always refresh and compare against the version that was actually committed.

Reply:
I see, changed.